addons/delivery/tests/test_delivery_cost.py

```diff
@@ -327,3 +327,52 @@ def test_add_carrier_on_picking(self):
             {'product_id': self.product_2.id, 'is_delivery': False, 'product_uom_qty': 1, 'qty_delivered': 1},
             {'product_id': self.normal_delivery.product_id.id, 'is_delivery': True, 'product_uom_qty': 1, 'qty_delivered': 0},
         ])
+
+
+    def test_delivery_cost_gift_card(self):
+        """
+        A customer has a carrier with the amount greater than the one to have
+        free shipping cost, then uses a gift card that lowers that amount to less
+        than the threshold: the shipping cost should still be 0.0
+        """
+        product_delivery_free = self.env['product.product'].create({
```

Comment:
It's not recommended but since we do not have a bridge module, this test should be skipped if `gift_card` is not installed.
```suggestion
        """
        if "gift.card" not in self.env:
            return
        product_delivery_free = self.env['product.product'].create({
```